src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -139,6 +139,21 @@ sealed class HashMap[A, +B] extends AbstractMap[A, B]
 
   override def par = ParHashMap.fromTrie(this)
 
+  /* Override to avoid tuple allocation in foreach */
+  private[collection] class HashMapKeys extends ImmutableDefaultKeySet {
+    override def foreach[U](f: A => U) = foreachEntry((key, _) => f(key))
+    override lazy val hashCode = super.hashCode()
+  }
+  override def keySet: immutable.Set[A] = new HashMapKeys
+
+  /** The implementation class of the iterable returned by `values`.
+   */
+  private[collection] class HashMapValues extends DefaultValuesIterable {
```

Comment:
> MiMa complaints seem ignorable.

Could these classes be private?  They can't escape into the API, for binary compatibility, anyways.

Reply:
I would like to be able to use them later for fast pathing the hashset operations like ++ as they share the same structure. We could make then private now though

Reply:
Ah, right.  Half of my motivation was wondering/hoping it removed the MiMa false positive, anyways.